app/src/main/java/edu/uc/groupProject/topten/ui/main/MainViewModel.kt

```diff
@@ -20,50 +23,153 @@ import edu.uc.groupProject.topten.Service.StrawpollService
 class MainViewModel : ViewModel() {
     var list:MutableLiveData<ArrayList<ListItem>> = MutableLiveData<ArrayList<ListItem>>()
     var listService: ListService = ListService()
+    private var firestore : FirebaseFirestore = FirebaseFirestore.getInstance()
+
+    init{
+        firestore.firestoreSettings = FirebaseFirestoreSettings.Builder().build()
+
+        writeListToDatabase()
+        waitForListUpdate()
+    }
 
     /**
-     * Passes the fetchList function down to the service layer.
-     * @return listService.fetchList(listItems)
+     * Writes the list to the Firebase Database. Currently writes the mocked data below to the
+     * database.
      */
-    fun fetchList(listName: String) {
-        list = listService.fetchList(listName)
+    fun writeListToDatabase(){
+        var testList = TestDTO("Top Fifteen Movies", false, true, true, Date())
+        var  listsReference = (firestore.collection("lists").document(testList.listName))
+
+        listsReference.set(testList).addOnSuccessListener {
+            Log.d("Firebase", "document saved");
+        }.addOnFailureListener{
+            Log.d("Firebase", "Save Failed");
+        }
+
+        var listItemCollectionReference = listsReference.collection("MyListItems")
+        var arrayOfListItemsToAdd : Array<ListItem> = createSampleListItems();
+
+        for(item in arrayOfListItemsToAdd){
+            listItemCollectionReference.document(item.title).set(item);
+        }
     }
 
-    fun fetchStrawpoll(id: Int): MutableLiveData<Strawpoll>? {
-        val service = StrawpollService()
-        return service.getStrawpoll(id)
+    /**
+     * Creates the mocked data list used by the program for testing purposes
+     * @return sampleListItems
+     */
+    fun createSampleListItems():Array<ListItem>{
+
+        var sampleListItems = arrayOf(
+            ListItem("The Dark Knight", "A movie about Batman", 100),
+            ListItem("The Return of the King", "A movie about a ring and some eagles", 150),
+            ListItem("The Empire Strikes Back", "A movie about some light wands and parent issues", 200),
+            ListItem("The Godfather", "n/a", 24),
+            ListItem("The Avengers", "They all team up to fight bad guys", 231),
+            ListItem("Inception", "", 12),
+            ListItem("E.T", "", 124),
+            ListItem("The Matrix", "", 42)
+        )
+        return sampleListItems
     }
-  
+
     /**
-     * This function fires every time a vote is added or changed on the generated list. Responsible
-     * for creating DTO objects from the database list.
+     * Provide the user name from the firebase database
+     * @return userName
      */
-    private var firestore : FirebaseFirestore = FirebaseFirestore.getInstance()
-    init{
```

Comment:
Moved init block to the top of the class . 
Link: https://kotlinlang.org/docs/coding-conventions.html#class-layout